File: ios/src_library/React Native/RCTContextMenu/RCTContextMenuManager.m

//
//  RCTContextMenuManager.m
//  nativeUIModulesTest
//
//  Created by Dominic Go on 7/14/20.
//

#import <React/RCTViewManager.h>
 

@interface RCT_EXTERN_MODULE(RCTContextMenuManager, RCTViewManager)

// -----------------------------------
// MARK: Props - RN Comp Props: Events
// -----------------------------------

RCT_EXPORT_VIEW_PROPERTY(onMenuShow        , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMenuHide        , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMenuCancel      , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPressMenuItem   , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPressMenuPreview, RCTBubblingEventBlock);

// -----------------------------------
// MARK: Props - RN Comp Props: Values
// -----------------------------------

RCT_EXPORT_VIEW_PROPERTY(menuConfig, NSDictionary);

@end

File: ios/src_library/React Native/RCTContextMenu/RCTContextMenuManager.swift
//
//  RCTContextMenuManager.swift
//  nativeUIModulesTest
//
//  Created by Dominic Go on 7/14/20.
//

import Foundation

@objc(RCTContextMenuManager)
class RCTContextMenuManager: RCTViewManager {
  
  override static func requiresMainQueueSetup() -> Bool {
    return true;
  };
  
  override func view() -> UIView! {
    let view = RCTContextMenuView(bridge: self.bridge);    
    return view;
  };
};
